word-search.py

```diff
@@ -351,7 +353,52 @@ def new_game():  #merged and renamed word_search() into new_game()
         gridWordList = generate_word_list(valid_words, grid)
 
         foundWords = []
-        currentScore = 0  
+        currentScore = 0
+
+        # Timer integaration
+        game_duration  = timer(timeroption)
+        start_time = time.time() if game_duration else None
+        print("\nGame Start! Find words in the grid.")
+    
+        if game_duration: # Timer active options 1, 2, 3
+            print (f"Game Timer: {game_duration} seconds")
+        else:
+            print("\nGame is untimed!")
+
+        # Game loop
+        while True:
+            # Timer logic
+            if game_duration:
+                elapsed_time = time.time() -  start_time
+                remaining_time = game_duration - int(elapsed_time)
+                if remaining_time <= 0: # Time's up
+                    print("Time's up. Game Over!")
+                    break
+
+            print(f"\rTime remaining: {remaining_time} seconds", end="", flush=True)
+            time.sleep(1)
```

Comment:
We're missing handling if the user chooses the untimed option. 🙂 I chose the untimed option and got this error when I was testing:
<img width="740" alt="Screenshot 2024-12-14 at 7 10 23 PM" src="https://github.com/user-attachments/assets/bad15f24-8397-4f1f-bb92-9b159ca42800" />

Reply:
Untimed mode issue addressed.

Reply:
Choosing untimed no longer throws an error. But let's remove the "Time Remaining" line please, since it's not applicable here
<img width="729" alt="Screenshot 2024-12-15 at 2 18 58 PM" src="https://github.com/user-attachments/assets/89761a44-6992-4cb3-8888-cd084197dbc4" />

Reply:
"Time remaining: None seconds" modified. Thank you